modules/pubmaticBidAdapter.js

```diff
@@ -498,38 +605,86 @@ export const spec = {
-                  newBid.height = bid.hasOwnProperty('h') ? bid.h : req.video.h;
-                  newBid.vastXml = bid.adm;
-                }
-              });
-            }
-            if (bid.ext && bid.ext.deal_channel) {
-              newBid['dealChannel'] = dealChannelValues[bid.ext.deal_channel] || null;
-            }
-
-            bidResponses.push(newBid);
-          });
+            utils.isArray(seatbidder.bid) &&
+            seatbidder.bid.forEach(bid => {
+              let newBid = {
+                requestId: bid.impid,
+                cpm: (parseFloat(bid.price) || 0).toFixed(2),
+                width: bid.w,
+                height: bid.h,
+                creativeId: bid.crid || bid.id,
+                dealId: bid.dealid,
+                currency: respCur,
+                netRevenue: NET_REVENUE,
+                ttl: 300,
+                referrer: utils.getTopWindowUrl(),
+                ad: bid.adm
+              };
+              let parsedRequest = JSON.parse(request.data);
+              if (parsedRequest.imp && parsedRequest.imp.length > 0) {
+                parsedRequest.imp.forEach(req => {
+                  if (bid.impid === req.id && req.hasOwnProperty('video')) {
+                    newBid.mediaType = 'video';
+                    newBid.width = bid.hasOwnProperty('w') ? bid.w : req.video.w;
+                    newBid.height = bid.hasOwnProperty('h') ? bid.h : req.video.h;
+                    newBid.vastXml = bid.adm;
+                  }
+                  if (bid.impid === req.id && req.hasOwnProperty('native')) {
+                    newBid.mediaType = 'native';
+                    newBid.native = {};
+                    if (bid.hasOwnProperty('adm')) {
+                      var adm = '';
+                      try {
+                        adm = JSON.parse(bid.adm);
+                      } catch (ex) {
+                        adm = JSON.parse(bid.adm.replace(/\\/g, ''));
+                      }
+                      if (adm && adm.native && adm.native.assets && adm.native.assets.length > 0) {
+                        for (let i = 0, len = adm.native.assets.length; i < len; i++) {
+                          switch (adm.native.assets[i].id) {
+                            case 1:
+                              newBid.native.title = adm.native.assets[i].title && adm.native.assets[i].title.text;
+                              break;
+                            case 2:
+                              newBid.native.image = {
+                                url: adm.native.assets[i].img && adm.native.assets[i].img.url,
+                                height: adm.native.assets[i].img && adm.native.assets[i].img.h,
+                                width: adm.native.assets[i].img && adm.native.assets[i].img.w,
+                              };
+                              break;
+                            case 3:
```

Comment:
move the hard-coded case ids to constants

Reply:
Done 
moved to constant